lib/eventsource_ex.ex

```diff
@@ -8,15 +8,27 @@ defmodule EventsourceEx do
     opts = Keyword.put(opts, :stream_to, parent)
     |> Keyword.put(:url, url)
 
-    GenServer.start(__MODULE__, opts, opts)
+    GenServer.start_link(__MODULE__, opts, opts)
   end
 
-  def init(opts \\ []) do
+  defp parse_options(opts) do
     url = opts[:url]
     headers = opts[:headers]
     parent = opts[:stream_to]
+    follow_redirect = opts[:follow_redirect]
+    ssl = opts[:ssl]
+    options = [stream_to: self(), ssl: ssl, follow_redirect: follow_redirect,
```

Comment:
```suggestion
    http_options = [stream_to: self(), ssl: ssl, follow_redirect: follow_redirect,
```
Renaming the var helps clarify why the list is being built.